Set incremental search for user search function

root8000 opened the pull request:
# WHAT
Set incremental search for user search function
# WHY
ユーザー検索にてインクリメンタルサーチを実装するため

Commit:
Rename group.scss groups.scss

group.scssとgroups.scssが混在していたので、groups.scssに統一

Commit:
Modify group.rb

一時解除していたバリデーションを復活＆文字周りの細かな修正

Commit:
Modify users routes

ユーザー検索のインクリメンタルサーチ化のため、indexアクションを定義

Commit:
Set index action(users_controller)

ユーザー検索のインクリメンタルサーチ化のため、indexアクションを定義。
・keywordの前後で検索に引っかかるよう記述
・htmlとjsonの場合で処理を分けるよう記述

Commit:
Create index.json.jbuilder

Commit:
Create group.js

ユーザー検索のインクリメンタルサーチ化のため作成。
・検索フォームに入力される度に発火するイベントを作成
・追加、削除でメンバーの編集ができるイベントを作成

Commit:
Modify _form.html.haml

ユーザー検索のインクリメンタルサーチ化のため、追記。

Commit:
Modify index.html.haml

文字周りの細かな修正

MrUrakou reviewed (commented):
インクリメントしてる挙動を確認したいので、GIFで動画を貼って、再提出お願いします。

Commit:
Add comment for group.js

全体の流れの確認のため、コメント挿入

Commit:
Modify groups_controller

グループ新規作成後のリダイレクト先を、rootから新規作成したグループのチャット画面に変更

Commit:
Modify group.js

フォームにバックスペース等を入力した際、ユーザー一覧が表示されてしまう問題を、入力内容が空かどうかで非同期通信するか分岐するよう追記することで解決。

root8000 commented:
一通り実装できました！
![default](https://user-images.githubusercontent.com/35862581/36527611-b8132fe0-17f5-11e8-80fe-81e269dc0f1e.gif)

root8000 commented:
ただ、チャットメンバーの入力フォームでバックスペース等が入力されるとユーザー一覧が表示されてしまう問題の解決に、if文で「入力内容が空でなければAjaxへ、空であれば通信せずに一覧だけ消してね」という条件分岐を書いたところ、一見うまくいくのですが、何かしらのフォーム入力をした後、フォームの中身を全て消すと、以降keyupイベントが動かなくなる不具合が出てしまいました。リロードすれば再度動きます。
（伝わりますかね？？わかりづらくてすみません！！！）
![default](https://user-images.githubusercontent.com/35862581/36527686-06b4c348-17f6-11e8-9d86-5276ca88dde8.gif)
大変気になる不具合なのですが、もしここまでの実装でもぎりぎりOKであれば、「バックスペース等の入力によってフォームにユーザー一覧が出てしまう問題」の解決前でひとまずマージさせて頂けませんでしょうか？
最終課題のチームが私含め２人でして、もう１人の方をかなりお待たせしてしまっているので、なるだけ早くチャットスペースの開発を完了いたしたく…。

Commit:
Modify group.js

フォームに一度入力した後に入力内容を消すと、リロードしないとkeyupイベントが再発火しない現象の解決。検索一覧そのものを消すremoveを記述してしまっていたが、中身だけを消すemptyに書き換えたところ解決。

root8000 commented:
上記のイベントが再発火しないエラー、解決しました！
入力が空だった場合、ユーザー検索一覧に対して誤ってremoveを使っていたため、検索一覧そのものを消してしまっていました。検索一覧の中身だけ消すようemptyに書き換えたところ、無事解決しました。

root8000 commented:
上記のイベントが再発火しないエラー、解決しました！
入力が空だった場合、ユーザー検索一覧に対して誤ってremoveを使っていたため、検索一覧そのものを消してしまっていました。検索一覧の中身だけ消すようemptyに書き換えたところ、無事解決しました。
